### PR TITLE
feat(nats): use queues for processing nats messages

### PR DESCRIPTION
**Pull request by hejkerooo**
Motivation: https://docs.nats.io/using-nats/developer/receiving/queues

### CODE REVIEW

**Review by Harasz — APPROVED** [review]

File: src/modules/claim/claim.service.ts [hunk 1]
```diff
@@ -51,10 +51,16 @@ export class ClaimService {
...
 
   private initializeExchangeListener() {
-    this.nats.connection.subscribe(`*.${NATS_EXCHANGE_TOPIC}`, async data => {
-      this.logger.debug(`Got message ${data}`);
-      await this.claimQueue.add('save', data);
-    });
+    this.nats.connection.subscribe(
+      `*.${NATS_EXCHANGE_TOPIC}`,
+      {
+        queue: 'iam_cache_server_exchange_topic',
```
Comment (jrhender) [inline src/modules/claim/claim.service.ts:57]: @hejkerooo where is this queue name from? Is this configured in NATS?

File: src/modules/claim/claim.service.ts [hunk 2]
```diff
@@ -51,10 +51,16 @@ export class ClaimService {
...
+      },
+      async data => {
+        this.logger.debug(`Got message ${data}`);
+        await this.claimQueue.add('save', data);
+      },
+    );
```
Comment (jrhender) [inline src/modules/claim/claim.service.ts:63]: @Passerino with our messaging changes, I don't think we expect the cache-server to be receiving any messages at this moment, right? All communication **to** the cache-server is done via HTTP, I believe. Perhaps we can remove this code?

**Review by jrhender — COMMENTED** [review]
Some additional questions

File: src/modules/claim/claim.service.ts (hunk 1 above)
Comment (jrhender) [reply src/modules/claim/claim.service.ts:57]: @hejkerooo, thanks for the change btw, I didn't know about NATS queues, this is pretty awesome

File: src/modules/claim/claim.service.ts (hunk 2 above)
Comment (jrhender) [reply src/modules/claim/claim.service.ts:63]: @Passerino  Ah, nevermind, I see you have commented it out in your PR already, so I think we can remove it in your PR 😄  https://github.com/energywebfoundation/iam-cache-server/pull/205/files

**Pull request merged**